develop/tutorials/articles/03-portlets/packaging-a-jsf-application.markdown

```diff
@@ -131,65 +166,23 @@ deployment:
         </security-constraint>
     </web-app>
 
-## Auto-Generated WAB [](id=auto-generated-wab)
-
-The source of an auto-generated JSF Portlet WAB also follows the standard
-directory layout for WAR projects. However, it does *NOT* include a `bnd.bnd`
-file and also does *NOT* include a Java class annotated with `@Component`:
-
-- `src/`
-    - `main/`
-        - `java/`
-        - `webapp/`
-            - `WEB-INF/`
-                - `faces-config.xml`
-                - `liferay-display.xml`
-                - `liferay-plugin-package.properties`
-                - `liferay-portlet.xml`
-                - `portlet.xml`
-                - `web.xml`
-	 			
-**Benefits:**
-
-- Processed by the Liferay auto-deploy process, which adds the aforementioned
-  `PortletServlet` and `PluginContextListener` to the `WEB-INF/web.xml`
-  descriptor.
-- Processed by the Liferay.
-  [WAB Generator](https://github.com/liferay/liferay-portal/tree/master/modules/apps/foundation/portal-osgi-web/portal-osgi-web-wab-generator),
-  which automatically creates an OSGi-ready `META-INF/MANIFEST.MF`.
-- Can affect the content of `META-INF/MANIFEST.MF` by putting BND directives and
-  OSGi headers into the `WEB-INF/liferay-plugin-package.properties` file.
-
-**Drawbacks:**
-
-- Can't supply the `bnd.bnd` and can't utilize a build-time plugin such as the
-  [bnd-maven-plugin](http://njbartlett.name/2015/03/27/announcing-bnd-maven-plugin.html).
-- Can't take advantage of OSGi Declarative Services (DS).
-- Developer must specify the `WEB-INF/portlet.xml` and
-  `WEB-INF/liferay-portlet.xml` descriptors.
+$$$
 
-+$$$
+## Utilizing OSGi Services [](id=utilizing-osgi-services)
 
-**Note:** To see the output of the Liferay WAB Generator, you can override the
-following properties:
+Both approaches can use custom-built OSGi services, since both result in the deployment of a WAB (true OSGi modules). Also, both approaches can be used take advantage of OSGi Declarative Services via the `@Reference` annotation:
```

Comment:
The UserLocalService that is specified below is an out-of-the-box OSGi service, not a custom-built service. So I recommend "Both approaches can utilize custom OSGi services and/or out-of-the-box services provided by Liferay Portal."